### PR TITLE
feat: Add soft delete for transactions

### LINKED ISSUE #312: Soft delete functionality improvements
Context: https://github.com/interledger/testnet/pull/292#discussion_r1176325628

User is able to delete the payment pointer, but right now there is a soft delete happening, basically in the DB the PP becomes inactive. When creating a PP with the same name as the deleted one, the previous transactions will be assigned to the new one. This is not desired, so we will add a flag for transactions, that they are old transactions-deprecated, and we will not display them anymore.
We need to improve on the PP delete functionality.
 When the user creates a new PP but with the same name as the one previously deleted, then the transactions done in the past will not appear anymore.

### PR DESCRIPTION
**Pull request by rico191013**
### Context

- fixes #312

### CODE REVIEW

File: packages/wallet/backend/migrations/20230728101910_update_transactions_columns.js
Comment (dragosp1011) [inline packages/wallet/backend/migrations/20230728101910_update_transactions_columns.js:11]: it should be a separate migration

Comment (dragosp1011) [reply packages/wallet/backend/migrations/20230728101910_update_transactions_columns.js:11]: also a new migration should be added to mark as deleted the transactions linked to existent inactive wallet addresses

File: packages/wallet/backend/src/transaction/service.ts
Comment (dragosp1011) [inline packages/wallet/backend/src/transaction/service.ts:91]: as id is the primary key, curious why we have this check here
I think it should be deleted

File: packages/wallet/backend/src/walletAddress/service.ts
Comment (dragosp1011) [inline packages/wallet/backend/src/walletAddress/service.ts:294]: it should be updated by wallet address id
there can be multiple wallet addresses in one account and it is not correct to mark as DELETED transactions that belongs to other wallet addresses that the one that is marked as inactive

**Review by dragosp1011 — CHANGES_REQUESTED** [review]

File: packages/wallet/backend/src/walletAddress/service.ts
Comment (Tymmmy) [inline packages/wallet/backend/src/walletAddress/service.ts:292]: remove console log

File: packages/wallet/backend/migrations/20230728101910_update_transactions_columns.js
Comment (dragosp1011) [reply packages/wallet/backend/migrations/20230728101910_update_transactions_columns.js]: @rico191013 the transactions linked to inactive wallet addresses should be marked as DELETED in the migration

**Review by beniaminmunteanu — APPROVED** [review]
lgtm, but wait for dragos approval

File: packages/wallet/backend/migrations/20240215061440_add_delete_flag_to_transactions_status_check_constraint.js
Comment (dragosp1011) [inline packages/wallet/backend/migrations/20240215061440_add_delete_flag_to_transactions_status_check_constraint.js:16]: a similar query should be added in `down` to restore the data.

Comment (dragosp1011) [reply packages/wallet/backend/migrations/20240215061440_add_delete_flag_to_transactions_status_check_constraint.js:16]: @beniaminmunteanu @Tymmmy @raducristianpopa 
I think it is not the best to add a new status in this case as the previous one is overwritten.
Keeping the status as it is and adding a separate field to indicate that transactions is deleted/inactive allows to have the correct transactions status for history and can be easily restored.

Comment (Tymmmy) [reply packages/wallet/backend/migrations/20240215061440_add_delete_flag_to_transactions_status_check_constraint.js:16]: Yes, I agree, we need a new field here to indicate that it is an inactive PP

Comment (beniaminmunteanu) [reply packages/wallet/backend/migrations/20240215061440_add_delete_flag_to_transactions_status_check_constraint.js:16]: good point.

File: packages/wallet/backend/migrations/20240215061440_add_active_column.js
Comment (dragosp1011) [inline packages/wallet/backend/migrations/20240215061440_add_active_column.js:7]: replace with `deletedAt` column that will keep the time of deletion

Comment (rico191013) [reply packages/wallet/backend/migrations/20240215061440_add_active_column.js:7]: How can I detect the time of deletion of old transactions?

Comment (dragosp1011) [reply packages/wallet/backend/migrations/20240215061440_add_active_column.js:7]: use `updatedAt` from wallet address

**Review by dragosp1011 — DISMISSED** [review]

**Pull request merged**
